### PR TITLE
Default to creating 1Inch swap without splitting

### PR DESCRIPTION
**Pull request by nlordell**
This PR configures the 1Inch solver make more gas efficient swaps by:
- To use a maximum of 1 swap "path"
- Setting a limit on gas prices
- Only allowing a single connector token.

### Test Plan

Added a test around new `Amount` type, previous tests continue to pass, including:
```
$ cargo test -- --ignored --nocapture solve_order_on_one
...
Spender {
    address: 0x11111112542d85b3ef69ae05771c2dccff4faa26,
}
test oneinch_solver::api::tests::oneinch_spender_address ... ok
Swap {
...
}
test oneinch_solver::api::tests::oneinch_swap_without_amount_checks_and_splitting ... ok
Swap {
...
}
test oneinch_solver::api::tests::oneinch_swap ... ok
Settlement {
...
}
test oneinch_solver::tests::solve_order_on_oneinch ... ok
...
```

### CODE REVIEW

File: solver/src/oneinch_solver/api.rs [hunk 1]
```diff
@@ -48,6 +48,36 @@ impl Display for Slippage {
     }
 }
 
+/// Parts to split a swap.
+///
+/// This type is generic on the maximum number of splits allowed.
+#[derive(Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub struct Parts<const N: usize>(usize);
+
+impl<const N: usize> Parts<N> {
```
Comment (fleupold) [inline solver/src/oneinch_solver/api.rs:57]: So fancy, monsieur Const Generics 😄

**Review by fleupold — APPROVED** [review]
Very nice change. I was looking at their API this morning and wondering if we should also set the gasLimit to 750k (instead of checking it after the fact) and maybe reduce the complexity level to 1?

File: solver/src/oneinch_solver/api.rs (hunk 1 above)
Comment (nlordell) [reply solver/src/oneinch_solver/api.rs]: They are overkill, but I've been DYING to use them :stuck_out_tongue:.

**Comment by nlordell** [comment]
@fleupold I added gas limit and complexity level to this PR and updated the description accordingly.

**Review by vkgnosis — APPROVED** [review]

**Review by fleupold — APPROVED** [review]
Great, I'll cut a new release together with #583 once they have been merged to staging (and tested for a few hours).

**Pull request merged**